kratos/solving_strategies/strategies/solving_strategy.h

```diff
@@ -276,15 +276,22 @@ class SolvingStrategy
         KRATOS_TRY
 
         if (GetModelPart().NodesBegin()->SolutionStepsDataHas(DISPLACEMENT_X) == false)
-            KRATOS_THROW_ERROR(std::logic_error, "It is impossible to move the mesh since the DISPLACEMENT var is not in the model_part. Either use SetMoveMeshFlag(False) or add DISPLACEMENT to the list of variables", "");
+	{
+            KRATOS_ERROR << "It is impossible to move the mesh since the DISPLACEMENT var is not in the model_part. Either use SetMoveMeshFlag(False) or add DISPLACEMENT to the list of variables" << std::endl;
+    	}
 
-        for (ModelPart::NodeIterator i = GetModelPart().NodesBegin();
-                i != GetModelPart().NodesEnd(); ++i)
-        {
-            (i)->X() = (i)->X0() + i->GetSolutionStepValue(DISPLACEMENT_X);
-            (i)->Y() = (i)->Y0() + i->GetSolutionStepValue(DISPLACEMENT_Y);
-            (i)->Z() = (i)->Z0() + i->GetSolutionStepValue(DISPLACEMENT_Z);
-        }
+        NodesArrayType& pNode = GetModelPart().Nodes();
+	auto numNodes = pNode.end() - pNode.begin();
```

Comment:
don't use auto for simple types.

this should be

     int numNodes = static_cast<int>(GetModelPart().Nodes().size());